lib/types.ts

```diff
@@ -385,7 +385,7 @@ type DeepModify<T> =
 /** Remove keys with `never` value from object type */
 export type NonNever<T extends {}> = Pick<T, { [K in keyof T]: T[K] extends never ? never : K }[keyof T]>;
 
-export type NonEmptyObject<T extends {}> = keyof T extends never ? never : T;
+export type NonEmptyObject<T extends Record<PropertyKey, any>> = keyof T extends never ? never : T;
```

Comment:
`PropertyKey` is not friendly for `keyofStringsOnly` option in tsconfig.json so we implemented our own

There's also a link attached to it, please have a look at it for more information – https://devblogs.microsoft.com/typescript/announcing-typescript-2-9-2/#support-for-symbols-and-numeric-literals-in-keyof-and-mapped-object-types

```suggestion
export type NonEmptyObject<T extends Record<KeyofBase, any>> = keyof T extends never ? never : T;
```